pandas/core/dtypes/cast.py

```diff
@@ -262,7 +262,7 @@ def maybe_downcast_to_dtype(result: ArrayLike, dtype: str | np.dtype) -> ArrayLi
                 dtype = "int64"
             elif inferred_type == "datetime64":
                 dtype = "datetime64[ns]"
-            elif inferred_type == "timedelta64":
+            elif inferred_type in ["timedelta", "timedelta64"]:
```

Comment:
Off topic, but should the above elif include `"datetime"`?

Reply:
tried it and it broke things, so firmly in the "maybe" column